### PR TITLE
Generalise the filter search JS

### PR DESCRIPTION
**Pull request by JamieCleare2525**
## Context

- Make the filter search functionality use a more generic/shared JS.

## Changes proposed in this pull request

- Replace existing `app/javascript/controllers/claims_support_filter_search_controller.js` and `app/javascript/controllers/placements_filter_search_controller.js` with a generic `app/javascript/controllers/filter_search_controller.js` which can be used for all filter search functionality.

## Guidance to review

- Using the search bars on filters (with search bars) she reduce the number of options shown.
- When JS is disabled, the search bar show not display.

@DFE-Digital/track-pay 
- Changes have removed the persisting search inputs in the search bars. This was confirmed with @ollietreend using the following example:
  - Example from GOV.UK: https://www.gov.uk/search/policy-papers-and-consultations
They use ajax to update the page, so the state naturally remains because the page hasn't reloaded.
But then when you do reload the page, the search filter gets reset.

## Link to Trello card

https://trello.com/c/Q6o53Y9p/514-create-a-reusable-component-for-search-filters

## Screenshots

_Claims_
(With JS enabled)

https://github.com/DFE-Digital/itt-mentor-services/assets/17162488/5154e975-3ded-4a83-a7b8-469900ee070a

(With JS disabled)

Screenshot 2024-06-25 at 10 46 18

_Placements_
(With JS enabled)

https://github.com/DFE-Digital/itt-mentor-services/assets/17162488/7f5448dd-7519-4c67-a84d-c904dc4aa3cb

(With JS disabled)

Screenshot 2024-06-25 at 10 49 31

### CODE REVIEW

**Comment by JamieCleare2525** [comment]
@ollietreend I can't approve my own PR. But I approve the changes you have made.

**Review by ollietreend — APPROVED** [review]
This looks good to me. Let's get a review from @DFE-Digital/track-pay before merging.

**Review by CatalinVoineag — APPROVED** [review]

**Pull request merged**
